python/dolfinx/fem/linearproblem.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2020 Jørgen S. Dokken
+#
+# This file is part of DOLFINX (https://www.fenicsproject.org)
+#
+# SPDX-License-Identifier:    LGPL-3.0-or-later
+import typing
+
+import ufl
+from dolfinx import fem
+from petsc4py import PETSc
+
+
+class LinearProblem():
+    """
+    Class for solving a linear variational problem of the form a(u,v)=L(v)
+    using PETSc as a linear algebra backend.
+    """
+    _globalcount = 0
```

Comment:
What is this for?

Reply:
As @michalhabera mentioned in a previous comment, to make the PETSc-options unique to the current solver (imagine having three `LinearProblem`s with different options, we need a unique prefix to the options. These are create using `         ufl.utils.counted.counted_init(self, countedclass=LinearProblem)` which is the class used to give for instance `ufl.Coefficient` a unique id.

Reply:
I've pushed a simpler fix - just uses `id(self)`.